src/view/reviewManager.ts

```diff
@@ -865,235 +493,31 @@ export class ReviewManager implements vscode.DecorationProvider {
-					let matchedFile: GitFileChangeNode;
-					if (matchedFiles && matchedFiles.length) {
-						matchedFile = matchedFiles[0];
-
-						let contentDiff: string;
-						if (document.isDirty) {
-							const documentText = document.getText();
-							const details = await this._repository.getObjectDetails(this._lastCommitSha!, matchedFile.fileName);
-							const idAtLastCommit = details.object;
-							const idOfCurrentText = await this._repository.hashObject(documentText);
-
-							// git diff <blobid> <blobid>
-							contentDiff = await this._repository.diffBlobs(idAtLastCommit, idOfCurrentText);
-						} else {
-							// git diff sha -- fileName
-							contentDiff = await this._repository.diffWith(this._lastCommitSha!, matchedFile.fileName);
-						}
-
-						matchingComments = this._comments.filter(comment => comment.path! === fileName);
-						matchingComments = mapCommentsToHead(matchedFile.diffHunks, contentDiff, matchingComments);
-
-						let diffHunks = matchedFile.diffHunks;
-
-						for (let i = 0; i < diffHunks.length; i++) {
-							let diffHunk = diffHunks[i];
-							let start = mapOldPositionToNew(contentDiff, diffHunk.newLineNumber);
-							let end = mapOldPositionToNew(contentDiff, diffHunk.newLineNumber + diffHunk.newLength - 1);
-							if (start > 0 && end > 0) {
-								ranges.push(new vscode.Range(start - 1, 0, end - 1, 0));
-							}
-						}
-					}
-
-					return {
-						threads: this.workspaceLocalCommentsToCommentThreads(matchedFile!, matchingComments, vscode.CommentThreadCollapsibleState.Collapsed),
-						commentingRanges: ranges,
-						inDraftMode: await this._prManager.inDraftMode(this._prManager.activePullRequest!)
-					};
-				}
+		this._localToDispose.push(this._reviewDocumentCommentProvider.onDidChangeWorkspaceCommentThreads(e => {
+			this._onDidChangeWorkspaceCommentThreads.fire(e);
+		}));
 
-				return;
-			},
-			createNewCommentThread: this.createNewCommentThread.bind(this),
-			replyToCommentThread: this.replyToCommentThread.bind(this),
-			editComment: this.editComment.bind(this),
-			deleteComment: this.deleteComment.bind(this),
-			startDraft: supportsGraphQL ? this.startDraft.bind(this) : undefined,
-			deleteDraft: supportsGraphQL ? this.deleteDraft.bind(this) : undefined,
-			finishDraft: supportsGraphQL ? this.finishDraft.bind(this) : undefined,
-			startDraftLabel: 'Start Review',
-			deleteDraftLabel: 'Delete Review',
-			finishDraftLabel: 'Submit Review'
+		this._localToDispose.push(this._reviewDocumentCommentProvider.onDidChangeComments(comments => {
+			this._comments = comments;
+			this._onDidChangeDecorations.fire();
 		}));
 
 		this._localToDispose.push(vscode.workspace.registerWorkspaceCommentProvider({
 			onDidChangeCommentThreads: this._onDidChangeWorkspaceCommentThreads.event,
 			provideWorkspaceComments: async (token: vscode.CancellationToken) => {
```

Comment:
`registerWorkspaceCommentProvider` and `_onDidChangeWorkspaceCommentThreads` are the only lefovers, we can move them into `ReviewDocumentCommentProvider` as well.